### PR TITLE
Add support for Bolt 5

### PR DESCRIPTION
**Pull request by injectives**

### CODE REVIEW

File: driver/src/main/java/org/neo4j/driver/internal/messaging/common/CommonValueUnpacker.java [hunk 1]
```diff
@@ -417,6 +418,14 @@ private static ZonedDateTime newZonedDateTime( long epochSecondLocal, long nano,
...
-}
 
+    protected int getNodeFields()
+    {
+        return NODE_FIELDS;
+    }
 
+    protected int getRelationshipFields()
+    {
+        return 5;
```
Comment (gjmwoods) [inline driver/src/main/java/org/neo4j/driver/internal/messaging/common/CommonValueUnpacker.java]: nit: shall we go constant here too?

File: driver/src/main/java/org/neo4j/driver/types/Entity.java [hunk 2]
```diff
@@ -28,12 +28,22 @@
...
+     * unique identifier is a better choice.
      *
      * @return the id of this entity
+     * @deprecated superseded by {@link #elementId()}
      */
+    @Deprecated
     long id();
+
+    /**
+     * A unique id for this Entity. Ids are guaranteed to remain stable for the duration of the session they were found in, but may be re-used for other
```
Comment (gjmwoods) [inline driver/src/main/java/org/neo4j/driver/types/Entity.java]: Is the same still true for element Id?

**Review by gjmwoods — APPROVED** [review]
Looks great.

File: driver/src/main/java/org/neo4j/driver/types/Entity.java (hunk 2 above)
Comment (gjmwoods) [reply driver/src/main/java/org/neo4j/driver/types/Entity.java]: also shall we add a note to `id()` to say a value of `-1` indicates unsupported or something util we work it out

File: driver/src/main/java/org/neo4j/driver/internal/messaging/common/CommonValueUnpacker.java (hunk 1 above)
Comment (injectives) [reply driver/src/main/java/org/neo4j/driver/internal/messaging/common/CommonValueUnpacker.java]: Updated.

File: driver/src/main/java/org/neo4j/driver/types/Entity.java (hunk 2 above)
Comment (injectives) [reply driver/src/main/java/org/neo4j/driver/types/Entity.java]: Updated.

**Pull request merged**